File: takevolet app/scripts/seed_categories.dart
import 'package:supabase/supabase.dart';

void main() async {
  final supabase = SupabaseClient(
    'https://gfhmdpzmhakznuqhstrn.supabase.co',
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdmaG1kcHptaGFrem51cWhzdHJuIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODcyMjc3MTMsImV4cCI6MjEwMjgwMzcxM30.AnwbTfZNxU64QntSJVIsJoEqflIuOPqSSWs9CUb8-TE'
  );

  print('Deleting existing...');
  try {
    await supabase.from('build_categories').delete().neq('id', '00000000-0000-0000-0000-000000000000');
  } catch(e) {
    print('Delete failed, proceeding anyway... \$e');
  }

  final categories = [
    { 'main_category': 'service', 'sub_category': 'Architect', 'icon_name': 'architecture' },
    { 'main_category': 'service', 'sub_category': 'Civil Engineer', 'icon_name': 'engineering' },
    { 'main_category': 'service', 'sub_category': 'Contractor', 'icon_name': 'handyman' },
    { 'main_category': 'service', 'sub_category': 'Builder', 'icon_name': 'foundation' },
    { 'main_category': 'service', 'sub_category': 'Site Engineer', 'icon_name': 'architecture' },
    { 'main_category': 'service', 'sub_category': 'Mason', 'icon_name': 'handyman' },
    { 'main_category': 'service', 'sub_category': 'Electrician', 'icon_name': 'electrical_services' },
    { 'main_category': 'service', 'sub_category': 'Plumber', 'icon_name': 'plumbing' },
    { 'main_category': 'service', 'sub_category': 'Carpenter', 'icon_name': 'carpenter' },
    { 'main_category': 'service', 'sub_category': 'Painter', 'icon_name': 'format_paint' },
    { 'main_category': 'service', 'sub_category': 'Interior Designer', 'icon_name': 'chair' },
    { 'main_category': 'service', 'sub_category': 'Vasthu Checker', 'icon_name': 'explore' },
    { 'main_category': 'service', 'sub_category': 'Surveyor', 'icon_name': 'square_foot' },
    { 'main_category': 'service', 'sub_category': 'Borewell Operator', 'icon_name': 'water_drop' },
    { 'main_category': 'service', 'sub_category': 'Site Supervisor', 'icon_name': 'engineering' },
    
    { 'main_category': 'material', 'sub_category': 'Cement Supplier', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Steel Supplier', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Bricks Supplier', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Sand & Aggregates', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Electrical Supplier', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Plumbing Material', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Tiles & Flooring', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Paint Supplier', 'icon_name': 'category' },
    { 'main_category': 'material', 'sub_category': 'Wood & Glass Supplier', 'icon_name': 'category' },
    
    { 'main_category': 'transport', 'sub_category': 'JCB', 'icon_name': 'local_shipping' },
    { 'main_category': 'transport', 'sub_category': 'Bulldozer', 'icon_name': 'local_shipping' },
    { 'main_category': 'transport', 'sub_category': 'Crane', 'icon_name': 'local_shipping' },
    { 'main_category': 'transport', 'sub_category': 'Borewell Rig', 'icon_name': 'local_shipping' },
    { 'main_category': 'transport', 'sub_category': 'Lorry', 'icon_name': 'local_shipping' },
    { 'main_category': 'transport', 'sub_category': 'Tractor', 'icon_name': 'agriculture' },
    { 'main_category': 'transport', 'sub_category': 'Tipper', 'icon_name': 'local_shipping' },
    { 'main_category': 'transport', 'sub_category': 'Concrete Mixer', 'icon_name': 'local_shipping' },
    { 'main_category': 'transport', 'sub_category': 'Scaffolding', 'icon_name': 'category' },
    { 'main_category': 'transport', 'sub_category': 'Lifting Equipment', 'icon_name': 'category' },
    
    { 'main_category': 'project', 'sub_category': 'Residential', 'icon_name': 'house' },
    { 'main_category': 'project', 'sub_category': 'Commercial', 'icon_name': 'storefront' },
    { 'main_category': 'project', 'sub_category': 'Apartment', 'icon_name': 'apartment' },
    { 'main_category': 'project', 'sub_category': 'Villa', 'icon_name': 'villa' }
  ];

  for (var c in categories) {
    try {
      await supabase.from('build_categories').insert(c);
      print('Inserted ' + c['sub_category'].toString());
    } catch (e) {
      print('Error inserting ' + c['sub_category'].toString() + ': \$e');
    }
  }
  print('Done seeding!');
}

File: scripts/seed_categories.ts
import { PrismaClient } from '@prisma/client';
const prisma = new PrismaClient();

async function main() {
  await prisma.$executeRawUnsafe('DELETE FROM public.build_categories');
  const categories = [
    { m: 'service', s: 'Architect', i: 'architecture' },
    { m: 'service', s: 'Civil Engineer', i: 'engineering' },
    { m: 'service', s: 'Contractor', i: 'handyman' },
    { m: 'service', s: 'Builder', i: 'foundation' },
    { m: 'service', s: 'Site Engineer', i: 'architecture' },
    { m: 'service', s: 'Mason', i: 'handyman' },
    { m: 'service', s: 'Electrician', i: 'electrical_services' },
    { m: 'service', s: 'Plumber', i: 'plumbing' },
    { m: 'service', s: 'Carpenter', i: 'carpenter' },
    { m: 'service', s: 'Painter', i: 'format_paint' },
    { m: 'service', s: 'Interior Designer', i: 'chair' },
    { m: 'service', s: 'Vasthu Checker', i: 'explore' },
    { m: 'service', s: 'Surveyor', i: 'square_foot' },
    { m: 'service', s: 'Borewell Operator', i: 'water_drop' },
    { m: 'service', s: 'Site Supervisor', i: 'engineering' },
    
    { m: 'material', s: 'Cement Supplier', i: 'category' },
    { m: 'material', s: 'Steel Supplier', i: 'category' },
    { m: 'material', s: 'Bricks Supplier', i: 'category' },
    { m: 'material', s: 'Sand & Aggregates', i: 'category' },
    { m: 'material', s: 'Electrical Supplier', i: 'category' },
    { m: 'material', s: 'Plumbing Material', i: 'category' },
    { m: 'material', s: 'Tiles & Flooring', i: 'category' },
    { m: 'material', s: 'Paint Supplier', i: 'category' },
    { m: 'material', s: 'Wood & Glass Supplier', i: 'category' },
    
    { m: 'transport', s: 'JCB', i: 'local_shipping' },
    { m: 'transport', s: 'Bulldozer', i: 'local_shipping' },
    { m: 'transport', s: 'Crane', i: 'local_shipping' },
    { m: 'transport', s: 'Borewell Rig', i: 'local_shipping' },
    { m: 'transport', s: 'Lorry', i: 'local_shipping' },
    { m: 'transport', s: 'Tractor', i: 'agriculture' },
    { m: 'transport', s: 'Tipper', i: 'local_shipping' },
    { m: 'transport', s: 'Concrete Mixer', i: 'local_shipping' },
    { m: 'transport', s: 'Scaffolding', i: 'category' },
    { m: 'transport', s: 'Lifting Equipment', i: 'category' },
    
    { m: 'project', s: 'Residential', i: 'house' },
    { m: 'project', s: 'Commercial', i: 'storefront' },
    { m: 'project', s: 'Apartment', i: 'apartment' },
    { m: 'project', s: 'Villa', i: 'villa' }
  ];

  for (const c of categories) {
    await prisma.$executeRawUnsafe(
      'INSERT INTO public.build_categories (main_category, sub_category, icon_name) VALUES ($1, $2, $3)',
      c.m, c.s, c.i
    );
  }
  console.log('Done seeding!');
}
main();
